mono/tests/assemblyresolve_event6.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Reflection;
+
+public class App
+{
+	public static int Main ()
+	{
+		/* Regression test for #58950: We need for the
+		 * ReflectionOnlyAssemblyResolve event handler to throw an
+		 * exception.  The exception unwinds native code in the loader,
+		 * which leaves stale coop handles on the coop handle stack.
+		 * Then, the domain unload, below, will assert in
+		 * mono_handle_stack_free_domain (). */
+		var d = AppDomain.CreateDomain ("TestDomain");
+		var o = d.CreateInstanceAndUnwrap (typeof (App).Assembly.FullName, "App/Work") as Work;
+		var r = o.DoSomething ();
+		if (r != 0)
+			return r;
+		AppDomain.Unload (d);
+		return 0;
+	}
+
+	public class MyExn : Exception {
+		public MyExn () : base ("MyReflectionResolveEventHandler threw") {}
+	}
+
+	public class Work : MarshalByRefObject {
+		public Work () { }
+
+		public int DoSomething ()
+		{
+			AppDomain.CurrentDomain.ReflectionOnlyAssemblyResolve += new ResolveEventHandler (MyReflectionResolveEventHandler);
+			bool caught = false;
+			try {
+				Assembly a = Assembly.ReflectionOnlyLoadFrom ("assemblyresolve_asm.dll");
+				var t = a.GetType ("Asm2");
+				var m = t.GetMethod ("M"); // This triggers a load of TestBase
+				Console.WriteLine ("got '{0}'", m);
+			} catch (FileNotFoundException e) {
```

Comment:
Which one is supposed to be thrown? FNFE or MyExn?

Reply:
oops. thanks - should be FNFE - MyExn would be the leaky previous behavior.